packages/code-du-travail-api/src/server/routes/api.js

```diff
@@ -153,5 +155,34 @@ router.get(`${BASE_URL}/idcc`, async ctx => {
   });
 });
 
+/**
+ * Return annaire items matching the given departement.
+ *
+ * @example
+ * http://localhost:1337/api/v1/items/:id
+ *
+ * @param {string} :id The item ID to fetch.
+ * @returns {Object} Result.
+ */
+router.get(`${BASE_URL}/annuaire/search`, async ctx => {
+  let lon, lat;
+  let coordData = ctx.request.query.coord;
+  if (coordData) {
```

Comment:
Might want to separate the separate this in 2 functions

```js
const { coord, q } = ctx.request.query;
const [lon, lat] = await searchFromCoord(coord) || await searchFromQuery(q)
ctx.body = await cdtnAnnuaire.getItemsBydistance({ coord: { lon, lat } });

// [...]

async function searchFromCoord(coord) {
  if (!coord) {
    return false // or Promise.reject(false)
  }

  // [...]
}

async function searchFromQuery(query) {
  if (!query) {
    return false // or Promise.reject(false)
  }

  // [...]
}
```

Reply:
👍